cli/integrity/integrity_test.go

```diff
@@ -0,0 +1,103 @@
+package integrity_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+	"github.com/tarantool/tt/cli/integrity"
+)
+
+func TestNewSigner(t *testing.T) {
+	testCases := []struct {
+		name           string
+		privateKeyPath string
+	}{
+		{
+			name:           "Empty path",
+			privateKeyPath: "",
+		},
+		{
+			name:           "Arbitrary path",
+			privateKeyPath: "private.pem",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			signer, err := integrity.NewSigner(testCase.privateKeyPath)
+			require.Equal(t, err, nil, "did not expect an error")
+			require.NotNil(t, signer)
+		})
+	}
+}
+
+func TestRegisterWithIntegritySigner(t *testing.T) {
+	someStr := ""
+
+	testCases := []struct {
+		name    string
+		command *cobra.Command
+		dst     *string
+	}{
+		{
+			name:    "Empty command and dst",
+			command: nil,
+			dst:     nil,
+		},
+		{
+			name:    "Empty command",
+			command: nil,
+			dst:     &someStr,
+		},
+		{
+			name:    "Empty dst",
+			command: &cobra.Command{},
+			dst:     nil,
+		},
+		{
+			name:    "Nothing empty",
+			command: &cobra.Command{},
+			dst:     &someStr,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			integrity.RegisterWithIntegrityFlag(testCase.command, testCase.dst)
+			if testCase.command != nil {
+				require.Equal(t, reflect.DeepEqual(*testCase.command, cobra.Command{}), true,
```

Comment:
It still does not clear what exactly the test tests.
```suggestion
				require.False(t, testCase.command.Flags().HasFlags())
```

Reply:
Why not? The intention is to check whether the the call to `RegisterWithIntegrityFlag` mutates the command.

Reply:
The idea of RegistryWithIntegrityFlag is modify a FlagSet. The argument type is chosen incorrectly if you need to test too much.

Reply:
> The idea of RegistryWithIntegrityFlag is modify a FlagSet/add a flag into a FlagSet. So the argument type is chosen incorrectly if you need to test too much.

That's fair. I'll refine the argument domain

Reply:
Fixed